src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -187,7 +212,7 @@ private BAMFileReader(final BlockCompressedInputStream compressedInputStream,
                           final SAMRecordFactory factory)
         throws IOException {
         mIndexStream = indexStream;
-        mIsSeekable = true;
+        mIsSeekable = null != indexStream;
```

Comment:
This may fix the immediate problem, but I'm concerned that the treatment of `mIsSeekable` is now inconsistent across constructors in this class. Eg., the constructor that takes a `BlockCompressedInputStream` and a `File` for the index sets `mIsSeekable` to true unconditionally, regardless of whether the index file is null. We should probably attempt to impose consistency across constructors here rather than leave the class in an inconsistent state.